neusomatic/python/sequencing_features.py

```diff
@@ -0,0 +1,279 @@
+    lseq = lseq.decode() if isinstance(lseq, bytes) else lseq
+    rseq = rseq.decode() if isinstance(rseq, bytes) else rseq
+
+    seq41_ref = lseq + ref_base + rseq
+    seq41_alt = lseq + first_alt + rseq
+
+    ref_counts = genome.count_repeating_bases(seq41_ref)
+    alt_counts = genome.count_repeating_bases(seq41_alt)
+
+    homopolymer_length = max(max(ref_counts), max(alt_counts))
+
+    # Homopolymer spanning the variant site:
+    ref_c = 0
+    alt_c = 0
+    for i in rseq:
+        if i == ref_base:
+            ref_c += 1
+        else:
+            break
+
+    for i in lseq[::-1]:
+        if i == ref_base:
+            ref_c += 1
+        else:
+            break
+
+    for i in rseq:
+        if i == first_alt:
+            alt_c += 1
+        else:
+            break
+
+    for i in lseq[::-1]:
+        if i == first_alt:
+            alt_c += 1
+        else:
+            break
+
+    site_homopolymer_length = max(alt_c + 1, ref_c + 1)
+
+    return homopolymer_length, site_homopolymer_length
+
+
+def somaticOddRatio(n_ref, n_alt, t_ref, t_alt, max_value=100):
+
+    # Odds Ratio just like VarDict's output
+    sor_numerator = n_alt * t_ref
+    sor_denominator = n_ref * t_alt
+    if sor_numerator == 0 and sor_denominator == 0:
+        sor = nan
+    elif sor_denominator == 0:
+        sor = max_value
+    else:
+        sor = min(sor_numerator / sor_denominator, max_value)
+
+    return sor
+
+def max_vocabularies(seq_length):
+    # According to:
+    # https://doi.org/10.1093/bioinformatics/18.5.679
+    # Assume 4 different nucleotides
+    counts = 0
+    k = 1
+    while k <= seq_length:
+        
+        if 4**k < (seq_length - k + 1):
+            counts = counts + 4**k
+        else:
+            counts = counts + (seq_length-k+1 + 1) * (seq_length-k+1 - 1 + 1)/2
+            break
+        
+        k += 1
+                
+    return counts
+
+
+
+def LC(sequence):
+    # Calculate linguistic sequence complexity according to
+    # https://doi.org/10.1093/bioinformatics/18.5.679
+    # Assume 4 different nucleotides
+    sequence = sequence.upper()
+    
+    if not 'N' in sequence:
+        
+        number_of_subseqs     = 0
+        seq_length            = len(sequence)
+        max_number_of_subseqs = max_vocabularies(seq_length)
+    
+        for i in range(1, seq_length+1):
+            
+            #max_vocab_1 = 4**i
+            #max_vocab_2 = seq_length - i + 1
+            set_of_seq_n = set()
+    
+            for n, nth_base in enumerate(sequence):
+                
+                if n+i <= len(sequence):
+                    sub_seq = sequence[n:n+i]
+                    set_of_seq_n.add( sub_seq )
+    
+                    # All possible unique subseqs obtained. Break away and go no further. 
+                    #if ( max_vocab_1 >= max_vocab_2 ) and ( len(set_of_seq_n) == max_vocab_2 ):
+                    #    break
+    
+            num_uniq_subseqs  = len(set_of_seq_n)
```

Comment:
@msahraeian This and the next line can be moved outside of the for loop. Since there is no conflict between sequences of different sizes, the result should be the same.